bigtable/admin.go

```diff
@@ -2175,3 +2184,255 @@ func (ac *AdminClient) UpdateBackup(ctx context.Context, cluster, backup string,
 	_, err := ac.tClient.UpdateBackup(ctx, req)
 	return err
 }
+
+// AuthorizedViewConf contains information about an authorized view.
+type AuthorizedViewConf struct {
+	TableID          string
+	AuthorizedViewID string
+
+	AuthorizedViewTypeConf AuthorizedViewTypeConf
+	DeletionProtection     DeletionProtection
+}
+
+// AuthorizedViewTypeConf contains information about the type of an authorized view.
+type AuthorizedViewTypeConf struct {
+	AuthorizedViewType AuthorizedViewType
+
+	SubsetView *SubsetViewConf
+}
+
+// AuthorizedViewType represents the type of an authorized view.
+type AuthorizedViewType int
+
+const (
+	// AuthorizedViewTypeUnspecified represents an unspecified authorized view type.
+	AuthorizedViewTypeUnspecified AuthorizedViewType = iota
+	// AuthorizedViewTypeSubsetView represents subset view type of an authorized view.
+	AuthorizedViewTypeSubsetView
+)
+
+func (av AuthorizedViewConf) proto() *btapb.AuthorizedView {
+	var avp btapb.AuthorizedView
+
+	switch dp := av.DeletionProtection; dp {
+	case Protected:
+		avp.DeletionProtection = true
+	case Unprotected:
+		avp.DeletionProtection = false
+	default:
+		break
+	}
+
+	switch avt := av.AuthorizedViewTypeConf.AuthorizedViewType; avt {
+	case AuthorizedViewTypeSubsetView:
+		avp.AuthorizedView = &btapb.AuthorizedView_SubsetView_{
+			SubsetView: av.AuthorizedViewTypeConf.SubsetView.proto(),
+		}
+	default:
+		break
+	}
+	return &avp
+}
+
+// SetSubsetView sets the AuthorizedViewType to AuthorizedViewTypeSubsetView and the subsetView pointer accordingly
+func (av *AuthorizedViewTypeConf) SetSubsetView(s SubsetViewConf) error {
+	av.AuthorizedViewType = AuthorizedViewTypeSubsetView
+	av.SubsetView = &s
+	return nil
+}
```

Comment:
I'm a little concerned here:
SubsetView is just one variant in a one-of and is a concrete type. In the future, when you add another type of authorized view, how will this be represented? As it stands you will have to go down the same path as protobuf-java, which is having parallel setters that clobber each other:

// sets the SubSetView
conf.SetSubsetView(...)
// clears SubsetView and sets newKindOfView
conf.SetNewKindOfView(...)

This is extremely surprising for endusers because setters dont usually share storage. This is has been a constant source of customer confusion with gc rules.

[golang protobuf](https://protobuf.dev/reference/go/go-generated/#oneof) actually takes pains to avoid this pitfall by creating an interface for the oneof field and then has concrete structs implement that interface for each variant. I think this should follow a similar pattern - create an interface for AuthorizedViewKind that has a private that only SubsetViewConf implements. Then in the future, you can create a NewKindConf that implements that interface.